### ISSUE TITLE
Recently added ENUM column SQL-error

### ISSUE THREAD

**Issue by apologetics** [typeorm/typeorm#559]
Hi, I'm trying to use newly created enum feature but there's a bug that occurs when trying to put new column to db.

```
@Column('enum', {
  name: 'role',
  enum: Roles,
  default: 'GUEST',
})
public role: Role = 'GUEST';
```

Where Roles is an array of strings and Role type is one of these string values, e.g. 'GUEST', 'ADMIN', etc.
Maybe, I'm doing it wrong, but it says:

`Error: ER_PARSE_ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'NOT NULL DEFAULT 'GUEST'' at line 1`

Removing default value results into the same error but with `'... near 'NOT NULL at line 1'`.

Moreover, IDEs are complaining that Column decorator is incompatible with the default syntax that was passing prior to the update. I tried cache invalidation but zero hope, it's still bursting with squigglies. Probably, that's because there's

```
export declare function Column(type: "enum", options?: ColumnCommonOptions & ColumnEnumOptions): Function;
```

in .d.ts and thus it eagers for that 'enum'. With the example above, where I use 'enum', IDE's happy.

Again, probably issue on my side only. I'll have a look at it deeper.

**Comment by zumazu** [comment]
For now, I just used class-validator package by @pleerock (`@IsIn(Roles)` in my case) and it fits my needs even though the column is currently a varchar.

```
@IsIn(Roles)
  @Column({
    type: 'varchar',
    length: 10,
    name: 'role',
    default: 'GUEST',
  })
  public role: Role = 'GUEST';
```

Hope it helps if someone has the same issue.

**Comment by pleerock** [comment]
Yeah its an issue with enum right now, thank you for the report. I havent added test for all column types yet, something I plan to do this or next week. Once I do it all column types should work good.

**Comment by AlexMesser** [comment]
fixed and released in `0.1.0-alpha.29` version
